### PR TITLE
Interproscan

### PR DESCRIPTION
**Pull request by MartinPippel**
replace local interproscan module (that is used in the functional annotation pipeline) with the NF-core module 
- this step can now run in parallel 
- cpus are set to 4 by default, which needs to tested and verified on real samples

### CODE REVIEW

File: subworkflows/functional_annotation/main.nf [hunk 1]
```diff
@@ -43,7 +43,10 @@ workflow FUNCTIONAL_ANNOTATION {
         GFF2PROTEIN.out.proteins.splitFasta( by: params.records_per_file, file: true ),
         blastdb_ch.map{ meta, db -> db }.collect()
     )
-    INTERPROSCAN( GFF2PROTEIN.out.proteins.splitFasta( by: params.records_per_file, file: true ) )
+    INTERPROSCAN( 
+        GFF2PROTEIN.out.proteins.splitFasta( by: params.records_per_file, file: true ),
+        out_text='tsv'
```
Comment (mahesh-panchal) [inline subworkflows/functional_annotation/main.nf:48]: This is strange. This is not valid nextflow. Is there a warning in the nextflow log?

**Review by mahesh-panchal — COMMENTED** [review]
I was hoping the temperamental behavior of Interproscan in Nextflow would disappear. I wonder if this is because of the Nextflow version.

File: subworkflows/functional_annotation/README.md [hunk 2]
```diff
@@ -65,7 +48,8 @@ These command line tool parameters can be changed by overriding the `ext.args` v
 ```nextflow
 process {
     withName: 'INTERPROSCAN' {
-        ext.args = '-f TSV --iprlookup --goterms -pa -dp -t p'
+        //
```
Comment (mahesh-panchal) [inline subworkflows/functional_annotation/README.md:51]: What is the comment here?

File: subworkflows/functional_annotation/main.nf (hunk 1 above)
Comment (MartinPippel) [reply subworkflows/functional_annotation/main.nf:48]: There is no warning. I was not sure how to set this parameter. I tested it and its doing whats supposed to be. What is the official way?

File: subworkflows/functional_annotation/README.md (hunk 2 above)
Comment (MartinPippel) [reply subworkflows/functional_annotation/README.md:51]: I need to add this 
// -dp is always set by default and -f is now controlled via value(out_ext)

File: subworkflows/functional_annotation/main.nf (hunk 1 above)
Comment (mahesh-panchal) [reply subworkflows/functional_annotation/main.nf:48]:
```suggestion
        'tsv'
```
Passing the string will implicitly convert it to value channel so Nextflow can use it.
I.e. this is equivalent to using
```nextflow
    INTERPROSCAN( 
        GFF2PROTEIN.out.proteins.splitFasta( by: params.records_per_file, file: true ),
        Channel.value('tsv')
    )
```

File: subworkflows/functional_annotation/README.md (hunk 2 above)
Comment (mahesh-panchal) [reply subworkflows/functional_annotation/README.md:51]:
```suggestion
```

File: config/nf-core-defaults.config [hunk 3]
```diff
@@ -29,6 +29,7 @@ process {
     }
     withLabel: process_long {
         time   = 20.h
+        cpus   = 4
```
Comment (mahesh-panchal) [inline config/nf-core-defaults.config:32]: Don't forget to move this to functional_annotation.config so it keeps the nf-core defaults the same as nf-core settings

File: config/test.config [hunk 4]
```diff
@@ -30,12 +30,13 @@ if ( params.subworkflow == 'functional_annotation' ) {
     process {
         withName: 'FUNCTIONAL_ANNOTATION:INTERPROSCAN' {
             ext.args = [
-                '-f TSV',
+                //'-f TSV',
```
Comment (mahesh-panchal) [inline config/test.config:33]: Just delete this line as it won't ever be needed because of the nf-core module

File: config/test.config [hunk 5]
```diff
@@ -77,6 +78,7 @@ process {
     }
     withLabel: process_long {
         time   = 1.h
+        cpus   = 2
```
Comment (mahesh-panchal) [inline config/test.config:81]: Remove this one as we want to keep it similar to nf-core settings.

**Comment by mahesh-panchal** [comment]
You also need to update the test config to use max 4 cores for interproscan. See failing actions.

**Review by mahesh-panchal — APPROVED** [review]

**Comment by mahesh-panchal** [comment]
Ready to merge?

**Pull request merged**